src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
You might be right, could be just me used to operations returning `true` on success (when declared as a bool). I just think its wrong to widely use an enum with three values where it almost everywhere only two values are used/checked (hence the alternative of using a different enum in the special cases; I guess one could argue if `eval.c/get_dict_tv` and `edit.c/ins_compl_add` expose the same three results or not)